shm/Alcohol.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Alcohol.hpp"
+
+#include <iostream>
+#include <typeinfo>
+
+Alcohol::Alcohol(const std::string& name, size_t amount, size_t basePrice, Time* time)
+    : Alcohol(name, amount, basePrice, MAX_POWER, time) {
+}
+
+Alcohol::Alcohol(const std::string& name, size_t amount, size_t basePrice, size_t power, Time* time)
+    : Cargo(name, amount, basePrice, time),
+      _power(power) {
+}
+
+size_t Alcohol::getPrice() const {
+    return _basePrice * (_power / MAX_POWER);
+}
+
+Cargo& Alcohol::operator+=(const size_t amount) {
+    if (_amount + amount > _maxAmount) {
+        std::cerr << "Not enought space for that cargo!\n";
+        return *this;
+    }
+    _amount += amount;
+    return *this;
```

Comment:
You are returning *this anyway. Maybe you can do some refactoring here.

Reply:
+1